app/routes/companyInterest/components/CompanyInterestList.tsx

```diff
@@ -84,11 +115,61 @@ class CompanyInterestList extends Component<Props, State> {
         },
       })
       .then(() => {
-        this.props.push(`/companyInterest?semesters=${clickedOption.id}`);
+        this.props.push(
+          `/companyInterest?semesters=${clickedOption.id}&event=${this.state.selectedEvent}`
+        );
       });
   };
+  handleEventChange = (clickedOption: EventOption): void => {
+    const { value } = clickedOption;
+    this.props
+      .fetch({
+        filters: {
+          events: value !== null ? value : null,
+        },
+      })
+      .then(() => {
+        this.props.push(
+          `/companyInterest?semesters=${this.props.selectedSemesterOption.id}&event=${clickedOption.value}`
+        );
+      });
+    this.setState({ selectedEvent: clickedOption.value });
+  };
+
+  componentDidUpdate(prevProps, prevStates) {
+    if (
+      this.props.selectedSemesterOption !== prevProps.selectedSemesterOption ||
+      this.state.selectedEvent !== prevStates.selectedEvent
+    ) {
+      this.setState({
+        generatedCSV: undefined,
+      });
+    }
+  }
+
+  async exportInterestList(event?: string) {
+    const blob = await fetch(
+      getCsvUrl(
+        this.props.selectedSemesterOption.year,
+        this.props.selectedSemesterOption.semester,
+        event
+      ),
+      {
+        headers: {
+          Authorization: `Bearer ${this.props.authToken}`,
+        },
+      }
+    ).then((response) => response.blob());
+    return {
+      url: URL.createObjectURL(blob),
+      filename: `Company-interests-${this.props.selectedSemesterOption.year}-${
+        this.props.selectedSemesterOption.semester
+      }${'asdfasdf'}.csv`,
```

Comment:
what is this??

Reply:
lol, also, please don't start the filename with a capital word ❤️ And, this could probably be in norwegian as well.